docs/source/markdown/guides/how_to/data/custom_data.md

```diff
@@ -0,0 +1,289 @@
+want to train a segmentation model that will be able to detect the
+abnormal regions in the test set.
+
+### With Normal and Abnormal Images
+
+We could use [Folder](../../reference/data/image/folder.md) datamodule to train a model on this dataset.
+
+:::::{dropdown} Code Syntax
+::::{tab-set}
+:::{tab-item} API
+:sync: label-1
+
+We could run the following python code to create the custom datamodule:
+
+```{literalinclude} ../../../../snippets/data/image/folder/segmentation/default.txt
+:language: python
+```
+
+The [Folder](../../reference/data/image/folder.md) datamodule will create training, validation, test and
+prediction datasets and dataloaders for us. We can access the datasets
+and dataloaders using the following attributes:
+
+```{literalinclude} ../../../../snippets/data/image/folder/segmentation/datamodule_attributes.txt
+:language: python
+```
+
+To check what individual samples from dataloaders look like, we can run
+the following command:
+
+```{literalinclude} ../../../../snippets/data/image/folder/segmentation/dataloader_values.txt
+:language: python
+```
+
+We could check the shape of the images and masks using the following
+commands:
+
+```python
+print(train_data["image"].shape)
+# torch.Size([32, 3, 256, 256])
+
+print(train_data["mask"].shape)
+# torch.Size([32, 256, 256])
+```
+
+Training the model is as simple as running the following command:
+
+```{literalinclude} ../../../../snippets/train/api/segmentation/model_and_engine.txt
+:language: python
+```
+
+where we train a Patchcore model on this custom dataset with default model parameters.
+:::
+
+:::{tab-item} CLI
+:sync: label-2
+Here is the CLI command to create the same custom datamodule:
+
+```{literalinclude} ../../../../snippets/config/data/image/folder/segmentation/cli/default.yaml
+:language: yaml
+```
+
+Assume that we have saved the above config file as `segmentation.yaml`. We could run the following CLI command to train a Patchcore model on above dataset:
+
+```bash
+anomalib train --data segmentation.yaml --model anomalib.models.Patchcore
+```
+
+:::
+
+::::
+:::::
+
+This example demonstrates how to create a segmentation dataset with
+normal and abnormal images. We could expand this example to create a
+segmentation dataset with only normal images.
+
+### With Only Normal Images
+
+There are certain cases where we only have normal images in our dataset
+but would like to train a segmentation model. Anomalib provides
+synthetic anomaly generation capabilities to create abnormal images from
+normal images. We could use the [Folder](../../reference/data/image/folder.md) datamodule to train a model on
+this dataset.
+
+:::::{dropdown} Code Syntax
+:icon: code
+
+::::{tab-set}
+:::{tab-item} API
+:sync: label-1
+
+We could run the following python code to create the custom datamodule:
+
+```{literalinclude} ../../../../snippets/data/image/folder/segmentation/normal.txt
+:language: python
+```
+
+As can be seen from the code above, we only need to specify the
```

Comment:
I think it would be good to explain in a bit more detail that we need the synthetic data for validation (normalization and thresholding).